skrub/_gap_encoder.py

```diff
@@ -389,8 +394,12 @@ def get_feature_names_out(
         for i in range(n_components):
             x = encoding[:, i]
             labels = vocabulary[np.argsort(-x)[:n_labels]]
-            topic_labels.append(labels)
-        topic_labels = [prefix + ", ".join(label) for label in topic_labels]
+            label = ", ".join(labels)
+            # Avoid having twice the same name for the different features
+            if label in topic_labels:
+                label += ", {:}".format(i)
+            topic_labels.append(label)
+        topic_labels = [prefix + label for label in topic_labels]
```

Comment:
I guess the prefix could be added as each label is processed in the loop above, too